core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -138,9 +138,11 @@ class ControllerChannelManager(controllerContext: ControllerContext, config: Kaf
 
   private def removeExistingBroker(brokerState: ControllerBrokerStateInfo) {
     try {
+      // shutdown the RequestSendThread before closing the NetworkClient to
+      // avoid the concurrent use of the non-threadsafe classes as described in KAFKA-4959
+      brokerState.requestSendThread.shutdown()
```

Comment:
+1
This looks correct to me. cc @ijuma (since he was the last one to touch this piece of code), although the statement ordering in the KAFKA-2440 is pretty much remained the same it was prior to that patch.